lms/services/rsa_key.py

```diff
@@ -17,6 +17,35 @@ def __init__(self, db, aes_service):
         self._db = db
         self._aes_service = aes_service
 
+    def rotate(self, target_keys: int):
+        """
+        Create `target_keys` new keys and mark existing ones as expired.
+
+        Keys have a live-cycle of:
+            active -> expired -> deleted
+
+        In flight request we signed with an active key will still be
+        verifiable as the public key endpoint query (get_all_public_jwks)
+        includes expired keys.
+
+        This method is meant to be called periodically by a celery task or similar.
+        """
+        # Delete any already expired keys
+        self._db.execute(delete(RSAKey).where(RSAKey.expired.is_(True)))
+
+        # Mark existing keys as expired
+        self._db.execute(
+            update(RSAKey).where(RSAKey.expired.is_(False)).values(expired=True)
+        )
+
+        # Create new keys
+        for _ in range(target_keys):
+            self.generate()
+
+        assert (
+            self._db.query(RSAKey).filter_by(expired=False).count() == target_keys
+        ), "The number of active RSAKey doesn't match the target"
```

Comment:
This is a bit odd. If we allow people to control it, why only have one correct answer? Why do we enforce it after we've done the work? Is this to get an error, but not actually do anything?

It seems like we should either:

* Allow the caller to control the number independently of how many there are
* Read the number from DB and use it directly to create new keys

Reply:
Added a comment, I just want stop everything if that assumption is not true.